### PR TITLE
8339030: frame::print_value_on(outputStream* st, JavaThread *thread) doesn't need thread argument

### PR DESCRIPTION
**Pull request by lmesnik**
Method
frame::print_value_on(outputStream* st, JavaThread *thread) doesn't need thread argument

it usually is called with nullptr as second arg except
JavaThread::trace_frames()
where it is called with this.

It seems that thread has never been used since 2007 so makes sense just to get rid of it.

Tested building all builds available in CI and running tier13

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8339030](https://bugs.openjdk.org/browse/JDK-8339030): frame::print_value_on(outputStream* st, JavaThread *thread) doesn't need thread argument (**Enhancement** - P4)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20721/head:pull/20721` \
`$ git checkout pull/20721`

Update a local copy of the PR: \
`$ git checkout pull/20721` \
`$ git pull https://git.openjdk.org/jdk.git pull/20721/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20721`

View PR using the GUI difftool: \
`$ git pr show -t 20721`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20721.diff">https://git.openjdk.org/jdk/pull/20721.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20721#issuecomment-2311336200)

### CODE REVIEW

File: src/hotspot/share/runtime/frame.hpp [hunk 1]
```diff
@@ -433,8 +433,8 @@ class frame {
   void describe_pd(FrameValues& values, int frame_no);
 
  public:
-  void print_value() const { print_value_on(tty,nullptr); }
-  void print_value_on(outputStream* st, JavaThread *thread) const;
+  void print_value() const { print_value_on(tty); }
+  void print_value_on(outputStream *st) const;
```
Comment (dholmes-ora) [inline src/hotspot/share/runtime/frame.hpp:437]: Nit: the * should be at the end of `outputStream` not the start of `st`.

**Review by dholmes-ora — APPROVED** [review]
Looks like it has been unused since [JDK-4894843](https://bugs.openjdk.org/browse/JDK-4894843) was integrated back in 2003. :)

Thanks for the clean up. One nit below.

File: src/hotspot/share/runtime/frame.hpp (hunk 1 above)
Comment (lmesnik) [reply src/hotspot/share/runtime/frame.hpp]: Thanks! I missed that IDE "helped" me here.

**Review by dholmes-ora — APPROVED** [review]

**Review by coleenp — APPROVED** [review]
This looks great.  Thanks for cleaning this up.

**Pull request closed without merging**